master/internal/api_tasks.go

```diff
@@ -35,6 +35,16 @@ var (
 	taskNotFound = status.Error(codes.NotFound, "task not found")
 )
 
+var pendStates = []taskv1.State{
+	taskv1.State_STATE_PULLING,
+	taskv1.State_STATE_STARTING,
+}
+
+var queueStates = []taskv1.State{
+	taskv1.State_STATE_PENDING,
```

Comment:
the fact that STATE_PENDING Is not in "pending state" list is certainly confusing. I wish we had better names for these.